includes/class-wc-stripe-webhook-handler.php

```diff
@@ -49,6 +49,13 @@ class WC_Stripe_Webhook_Handler extends WC_Stripe_Payment_Gateway {
 	 */
 	protected $deferred_webhook_action = 'wc_stripe_deferred_webhook';
 
+	/**
+	 * The order object being processed.
+	 *
+	 * @var WC_Order
+	 */
+	protected $resolved_order;
```

Comment:
I'm not a fan of this; setting this Order attribute everywhere to be used only when the hook `wc_stripe_webhook_processed` is registered seems weird to me. It's the kind of thing we might forget when adding code to handle a new `event_type`,

Also, not all `events` have an Order associated (ie, `account.updated`), as we add more features, we might have more of those..

We are already passing the "notification event" received from Stripe to the `wc_stripe_webhook_processed` hook. If needed, the order can be obtained from the notification payload object: (ie: `WC_Stripe_Helper::get_order_by_charge_id( $notification->data->object->charge )`)

Reply:
I am also not a big fan of this approach for the same reasons, but I understand Dale's point of view, too. It is like a "bad code" that is useful for merchants. I am not sure what the best course of action is here (keep it or remove it).

Reply:
I'd appreciate it if you could handle that, Dale 🙏 . Should I keep the `$resolved_order` logic here or revert it meanwhile?